### DISCUSSION TITLE
C# class generálás

### DISCUSSION THREAD

**Discussion by nalatreb** [nav-gov-hu/eVAT#114] [Kérdések és válaszok - Gépi interfész]
Sziasztok!

[xscgen](https://github.com/mganss/XmlSchemaClassGenerator/tree/master)-t használok és az olyan típusokhoz, amik valamilyen AtomicStringType{x}-ből származnak duplán generálja le a validációt. Például az **InvoiceDigestType** **currency** element-je, ami CurrencyType típusú.

Generált kód:
``` c#
        /// <summary>
        /// <para xml:lang="en">Currency of the invoice</para>
        /// <para xml:lang="en">Currency type (three digit ISO 4217 currency code)</para>
        /// <para xml:lang="en">Minimum length: 1.</para>
        /// <para xml:lang="en">Maximum length: 4.</para>
        /// <para xml:lang="en">Minimum length: 3.</para>
        /// <para xml:lang="en">Maximum length: 3.</para>
        /// <para xml:lang="en">Pattern: [A-Z]{3}.</para>
        /// </summary>
        [System.ComponentModel.DescriptionAttribute("Currency of the invoice")]
        [System.ComponentModel.DataAnnotations.MinLengthAttribute(1)]
        [System.ComponentModel.DataAnnotations.MaxLengthAttribute(4)]
        [System.ComponentModel.DataAnnotations.MinLengthAttribute(3)]
        [System.ComponentModel.DataAnnotations.MaxLengthAttribute(3)]
        [System.ComponentModel.DataAnnotations.RegularExpressionAttribute("[A-Z]{3}")]
        [System.Xml.Serialization.XmlElementAttribute("currency")]
        public string Currency { get; set; }
```

A CurrencyType-nál így van megfogalmazva a **restriction** rész.

``` xsd
<xs:restriction base="AtomicStringType4">
    <xs:length value="3"/>
    <xs:pattern value="[A-Z]{3}"/>
</xs:restriction>
```

Valószínűleg itt az a probléma, hogy az xscgen a **length** miatt és az örökölt **minLenght** **maxLength** miatt is rak rá egy validálást. Felmerült bennem, hogy a restriction öröklődés-nél nem lenne jobb ebben az esetben a **minLength**, **maxLength** restricinöket felülírni az xsd-ben?

Esetleg valaki ismer alternatívát a c# osztály generáláshoz(ami nem a microsoftos xsd tool :D )?

Üdvözlettel,
Berci

**Comment by u319169** [comment]
Tisztelt Nalatreb!
Az a kérdésem, hogy fennáll még a probléma? Lévén eléggé régi a bejegyzés.
Üdvözlettel: u319169
